pydicom/filereader.py

```diff
@@ -230,6 +233,15 @@ def data_element_generator(fp,
     element_struct_unpack = element_struct.unpack
     defer_size = size_in_bytes(defer_size)
 
+    tag_set = None
+    if specific_tags is not None:
+        tag_set = set()
```

Comment:
No, because if the set is empty, it shall not be used. I could of course ask each time for an empty tag_set, but that would give you no advantage.